### PR TITLE
Added identify CTEs multiple evaluation pattern

### PR DESCRIPTION
**Pull request by meghabedi**
Added Identify CTEs evaluate multiple times anti-pattern with following class files: 
1. IdentifyCTEsEvalMultipleTimes 
2. IdentifyCTEsEvalMultipleTimesVisitor 
3. IdentityCTEsEvalMultipleTimesTest

### CODE REVIEW

File: tools/zetasql-helper/bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/Main.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 import java.util.stream.Collectors;
 import org.apache.commons.cli.ParseException;
 
-public class Main {
+public class Mainw {
```
Comment (franklinWhaite) [inline tools/zetasql-helper/bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/Main.java]:
```suggestion
public class Main {
```

File: tools/zetasql-helper/bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/visitors/IdentifyCTEsEvalMultipleTimesVisitor.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class IdentifyCTEsEvalMultipleTimesVisitor extends ParseTreeVisitor {
+
+  // A string template to be used for generating the suggestion message.
+  private final String MULTIPLE_CTE_SUGGESTION_MESSAGE =
+      "Your query uses CTE alias %s which is referenced %d times.";
```
Comment (franklinWhaite) [inline tools/zetasql-helper/bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/visitors/IdentifyCTEsEvalMultipleTimesVisitor.java]:
```suggestion
      "CTE with multiple references: alias %s is referenced %d times.";
```

Comment (franklinWhaite) [reply tools/zetasql-helper/bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/visitors/IdentifyCTEsEvalMultipleTimesVisitor.java:16]: Could you please add the new anti pattern to the main [here](https://github.com/meghabedi/professional-services/blob/anti-patterns/tools/zetasql-helper/bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/Main.java#L50).

**Comment by franklinWhaite** [comment]
Could you please add the new anti pattern to the main [here](https://github.com/meghabedi/professional-services/blob/anti-patterns/tools/zetasql-helper/bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/Main.java#L50).

**Comment by meghabedi** [comment]
1. Made changes as per your suggestion. 
2. Added CTE recommendation to main.
3. Modified Test cases CTE_SUGGESTION_MESSAGE.

**Review by franklinWhaite — APPROVED** [review]

**Pull request merged**